master/buildbot/steps/source/mtn.py

```diff
@@ -216,52 +220,39 @@ def removeFiles(self, files):
                 return
         defer.returnValue(0)
 
-    def _clone(self, abandonOnFailure=False):
-        command = ['mtn', 'db', 'init', '--db', self.database]
-        d = self._dovccmd(command, abandonOnFailure=abandonOnFailure)
-        d.addCallback(lambda _: self._pull())
-        return d
-
     def _checkout(self, abandonOnFailure=False):
-        command = ['mtn', 'checkout', '.', '--db=%s' % (self.database)]
+        command = ['mtn', 'checkout', self.workdir, '--db', self.database]
         if self.revision:
             command.extend(['--revision', self.revision])
         command.extend(['--branch', self.branch])
+        return self._dovccmd('', command, abandonOnFailure=abandonOnFailure)
 
-        return self._dovccmd(command, abandonOnFailure=abandonOnFailure)
-
-    def _update(self, dopull=True, abandonOnFailure=False):
-        if dopull:
-            d = self._pull()
-        else:
-            d = defer.succeed(0)
-        command = ['mtn', 'update', '--db=%s' % (self.database)]
+    def _update(self, abandonOnFailure=False):
+        command = ['mtn', 'update']
         if self.revision:
             command.extend(['--revision', self.revision])
         else:
-            command.extend(["-r", "h:" + self.branch])
-        command.extend(["-b", self.branch])
-
-        d.addCallback(lambda _: self._dovccmd(command, abandonOnFailure=abandonOnFailure))
-        return d
+            command.extend(['--revision', 'h:' + self.branch])
+        command.extend(['--branch', self.branch])
+        return self._dovccmd(self.workdir, command,
+                             abandonOnFailure=abandonOnFailure)
 
     def _pull(self, abandonOnFailure=False):
-        command = ['mtn', 'pull', self.sourcedata,
-                   '--db=%s' % (self.database)]
+        command = ['mtn', 'pull', self.sourcedata, '--db', self.database]
         if self.progress:
             command.extend(['--ticker=dot'])
         else:
             command.extend(['--ticker=none'])
-        d = self._dovccmd(command, abandonOnFailure=abandonOnFailure)
+        d = self._dovccmd('', command, abandonOnFailure=abandonOnFailure)
```

Comment:
Okay, I changed _dovccmd to allow workdir to be passed as a keyword argument after the command (no default, though). Please go ahead and change RemoteShellCommand to match and to get rid of the exact same issue in steps/source/svn.py.

Plus it's now a '.' (dot) to make it a bit more obvious that it's a relative path.